compiler/rustc_ast_lowering/src/format.rs

```diff
@@ -20,10 +48,106 @@ impl<'hir> LoweringContext<'_, 'hir> {
         let mut fmt = Cow::Borrowed(fmt);
         if self.tcx.sess.opts.unstable_opts.flatten_format_args {
             fmt = flatten_format_args(fmt);
-            fmt = inline_literals(fmt);
+            fmt = self.inline_literals(fmt);
         }
         expand_format_args(self, sp, &fmt, allow_const)
     }
+
+    /// Try to convert a literal into an interned string
+    fn try_inline_lit(&self, lit: token::Lit) -> Option<Symbol> {
+        match LitKind::from_token_lit(lit) {
+            Ok(LitKind::Str(s, _)) => Some(s),
+            Ok(LitKind::Int(n, ty)) => {
+                // platform-dependent usize and isize MAX
+                let usize_bits = self.tcx.data_layout.pointer_size.bits();
+                let usize_max = if usize_bits >= 128 { u128::MAX } else { 1u128 << usize_bits - 1 };
+                let isize_max = usize_max >> 1;
```

Comment:
Perhaps just use the [`signed_int_max`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_abi/struct.Size.html#method.signed_int_max) and [`unsigned_int_max`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_abi/struct.Size.html#method.unsigned_int_max) methods?

Reply:
Fair.